api/document/serializers/content.py

```diff
@@ -71,6 +70,8 @@ def nest_annotations(annotations: Iterator[Annotation],
         # We're not allowing non-nested overlapping annotations, so we won't
         # compare ends when determining nesting
         while anote not in last:
+            if last.parent is None:
+                raise AssertionError('last.parent is None!')
```

Comment:
This is actually a potential bug that mypy may have caught. It's basically noticing that since `parent` is, according to the constructor, an _optional_ value, `last = last.parent` might make `last` `None`, which could then make `anote not in last` explode. Here I used type narrowing to ensure that `last.parent` isn't `None`, _but_ I also didn't know what to do if that happens, so I decided to just raise an `AssertionError` for now, since I assume this code wasn't designed to handle that kind of case, or else it'd crash during tests or in production?